### ISSUE TITLE
Config.prc/Confauto.prc papercuts

### ISSUE THREAD

**Issue by Moguri** [panda3d/panda3d#269]
Opening this issue to discuss some smaller annoyances around the Config.prc and Confauto.prc files. This is separate from the discussion in #245, which discusses the PRC system in general. Some points I would like to address:

* Rename files to start with a number (e.g., 00-confauto.prc and 01-config.prc). This makes maintaining alphabetical order for files easier without having to worry about the names themselves.
* Determine purpose of Config.prc and Confauto.prc
  * Should these be combined?
  * Should we be encouraging users to edit these? They are typically installed in system/library locations, and it may be better to encourage overriding vs editing.
* Move prc files from /etc/*.prc to /etc/panda3d/*.prc
* Can we remove some of the settings from either config file? There are some options that are set to their defaults, and they seem to be in the config file to provide information/documentation. Some settings that I think we can remove:
  * framebuffer-hardware (matches default and few people will want to change it)
  * framebuffer-software ( matches default and few people will want to change it)
  * use-movietexture (this option may not actually exist; I cannot find it in the codebase)
  * lock-to-one-cpu (commented out and few people will want to change it)
  * paranoid-clock (commented out and few people will want to change it)

**Comment by CFSworks** [comment]
> Rename files to start with a number (e.g., 00-confauto.prc and 01-config.prc). This makes maintaining alphabetical order for files easier without having to worry about the names themselves.

I agree with this. The old ppremake build system (and in-development CMake build system) both use(d) numeric prefixes just like that. CMake actually uses `20_panda.prc` for its autogenerated configuration.

> Determine purpose of Config.prc and Confauto.prc

I can't even tell you off the top of my head what the difference is. Confauto *seems* to contain all of the egg type tables used by the egg scenegraph generator, so it might make a little sense to make Confauto a "don't edit, will get overwritten by Panda updates" file and Config has the user-friendly options. (Maybe we could even get rid of Confauto and just make those the engine defaults altogether...?)

> Move prc files from /etc/.prc to /etc/panda3d/.prc

One thing I've never liked about Panda is how it defaults to looking in system paths for engine configuration. This seems like a holdover from Panda's "industrial VR installation" days, where it would actually run as a daemon on some SGI IRIX system hooked up to a DisneyQuest attraction full-time. It doesn't make much sense for an engine primarily intended for redistribution now. We should be trying to encourage loading prc out of Multifiles and/or the app's installation directory (if running as part of a released app) or the user's home directory (if running as part of the SDK), I think. (In the latter case - maybe some mechanism where a read-only default .prc is copied out of the Panda installation directory to the user's home directory on first launch?)

> Can we remove some of the settings from either config file? There are some options that are set to their defaults, and they seem to be in the config file to provide information/documentation.

Y'know... I'd be in favor of rewriting the default prc files from scratch. Especially after completing some of the tasks in #245, everything will have a different name anyway, and we should be setting a good example for newcomers to follow.

**Comment by Moguri** [comment]
@rdb and @CFSworks any objections to making these config settings the engine defaults and ditching Config.prc and Confauto.prc? Panda can still try to load these files if it finds them to avoid breaking existing setups.

**Comment by CFSworks** [comment]
You'll find no objections from me.

**Comment by rdb** [comment]
@Moguri No objections from me.  As far as I'm concerned we can move ahead with having the config files read according to the [XDG Base Directory spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) in 1.11.

I'm not entirely sure about removing `framebuffer-software/hardware`, it is occasionally useful to change this.

**Comment by rdb** [comment]
Sorry, I thought your question was regarding implementing what is described in the issue description, not regarding ditching Config.prc and Confauto.prc altogether.  I'm not sure about that; it is occasionally useful to unset things that are otherwise enabled by default.  We also still want to have defaults at least commented-out so that people see what they might want to modify.

**Comment by Moguri** [comment]
I do not like the idea of having global engine settings that we anticipate users will modify. These settings are also usually (always?) stored in a location that requires elevated privileges to edit. Instead, if a user wants to override engine defaults, I believe we should encourage users to add Panda3D configuration files to their user configuration directory (e.g., ~/.config/panda3d on Linux).

**Comment by serkkz** [comment]
I propose to make this part of the handmade. This will add a few lines to the user code and bring more clarity. It will also be a logical and consistent action. It also closes hack holes for users.
